File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {

    private static int m;
    private static boolean[] visited;

    public static void main(String[] args) throws IOException {
        init();
    }

    private static void init() throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        m = Integer.parseInt(input.readLine());
        visited = new boolean[21];
        StringBuilder answer = new StringBuilder();

        for (int i = 0; i < m; i++) {
            String[] s  = input.readLine().split(" ");
            findAnswer(s, answer);
        }

        System.out.println(answer);
    }


    private static void findAnswer(String[] s, StringBuilder answer) {
        if (s[0].equals("add")) {
            visited[Integer.parseInt(s[1])] = true;
            return;
        }

        if (s[0].equals("remove")) {
            visited[Integer.parseInt(s[1])] = false;
            return;
        }

        if (s[0].equals("check")) {
            if (!visited[Integer.parseInt(s[1])]) {
                answer.append("0").append("\n");
            } else {
                answer.append("1").append("\n");
            }
            return;
        }

        if (s[0].equals("toggle")) {
            if (!visited[Integer.parseInt(s[1])]) {
                visited[Integer.parseInt(s[1])] = true;
            } else {
                visited[Integer.parseInt(s[1])] = false;
            }
            return;
        }

        if (s[0].equals("all")) {
            for (int i = 1; i <= 20; i++) {
                visited[i] = true;
            }

            return;
        }
        if (s[0].equals("empty")) {
            for (int i = 0; i <= 20; i++) {
                visited[i] = false;
            }
        }
    }
}

File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>
using namespace std;
int m;
// 숫자는 1~20
// add는 x추가 있으면 무시
// remove x s에서 x제거 없으면 무시
// check x 있으면 1, 없으면 x
// toggle s에 x가 있으면 제거 없으면 x추가
// all -> S= 1,2,3...20
// empty -> 공집합
int S;
int main(){
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    cin >> m;
    for (int i = 0; i < m; i++){
        string s; int x;
        cin >> s;
        if (s != "all" && s != "empty") {
            cin >> x;
        }
        
        if (s == "add") {
            S |= (1 << x);
        } else if (s == "remove") {
            S&=~(1 << x);
        } else if (s == "check") {
            if (S & (1<<x)) {
                cout << '1' << '\n';
            } else {
                cout << '0' << '\n';
            }
        } else if (s == "toggle") {
            S^=(1<<x);
        } else if (s == "all") {
            S = (1 << 21) - 1;
        } else {
            // empty 
            S = 0;
        }

    }

    return 0;
}